File: quarto-semestre/AEDS 2/tps_resolvidos/tp01/ex2/palindromo.c
#include <stdio.h>
#include <string.h>

void verificarPalindromo(char frase[])
{
    int statusPalavra = 0;
    int tamanho = strlen(frase);

    for (int i = 0; i < tamanho; i++)
    {
        char letraComparada = frase[tamanho - i - 1];

        if (frase[i] == letraComparada)
        {
            statusPalavra = 1;
        }
        else
        {
            statusPalavra = 0;
            i = tamanho;
            ;
        }
    }

    if (statusPalavra == 1)
    {
        printf("SIM\n");
    }
    else
    {
        printf("NAO\n");
    }
}

int main()
{
    char frase[100];

    int condicao = 1;
    while (condicao == 1)
    {
        fgets(frase, sizeof(frase), stdin);
        frase[strcspn(frase, "\n")] = 0;

        if (strcmp(frase, "FIM") == 0)
        {
            condicao = 0;
        }

        verificarPalindromo(frase);
    }

    return 0;
}

File: quarto-semestre/AEDS 2/tps_resolvidos/tp01/ex1/Palindromo.java
import java.util.Scanner;

public class Palindromo {

    public static void verificarPalindromo(String frase) { // compara as letras da frase de forma que fiquem iguais "ao contrário"

        boolean statusPalavra = false;
        int tamanho = frase.length();

        for (int i = 0; i < tamanho; i++) {
            char letraComparada = frase.charAt(tamanho - i - 1); 

            if (frase.charAt(i) == letraComparada) { // aqui a comparação é feita
                statusPalavra = true;
            } else {
                statusPalavra = false;
                i = tamanho;
            }
        }

        if (statusPalavra == true) {
            System.out.println("SIM");
        } else {
            System.out.println("NAO");
        }
    }

    public static void main(String[] args) {
        String frase = "";
        Scanner scanner = new Scanner(System.in);

        frase = scanner.nextLine();

        while (!(frase.equals("FIM"))) {
            verificarPalindromo(frase);
            frase = scanner.nextLine();
        }

        scanner.close();
    }
}